docs/source/data/kedro_dataset_factories.md

```diff
@@ -0,0 +1,227 @@
+# Kedro dataset factories
+You can load multiple datasets with similar configuration using dataset factories, introduced in Kedro 0.18.12.
+
+The syntax allows you to generalise the configuration and reduce the number of similar catalog entries by matching datasets used in your project's pipelines to dataset factory patterns.
+
+## Generalise datasets with similar names and types
```

Comment:
The subheadings in the dataset factories docs were pretty long and made the table of contents on the left rather messy. I shortened them but would value input from @ankatiyar about whether the subheadings are now useful or too generic.